### ISSUE TITLE
Support for Opera?

### ISSUE THREAD

**Issue by AlexPoulsen** [beardedspice/beardedspice#96]
Could Opera support be added in a soon-ish update?

**Comment by Coder-256** [comment]
@Stillness-2 @1ps0
Since Opera does not support Applescript, this would be possible only with an extension + [native messaging](https://dev.opera.com/extensions/message-passing/#native-messaging).

This would be a **huge** undertaking (new target build required w/ STDIN open and able to communicate with XPC, manifest placing, per-user troubles app location moving/duplicates/updating, etc.), but would be the "right way" to do it and would work for not only Opera but also Chrome and probably all Chromium-based browsers.

**Comment by 1ps0** [comment]
@Coder-256 Interesting. This is definitely a good starting place for a 'doing it right' approach. 

The initial research is really helpful for the first steps forward. If you want, dig a bit deeper and maybe prototype some stuff to see how feasible it is. For major features like this it's always important to start small and even start separate. See how I did the [JS conversions discovery/prototyping](https://gist.github.com/1ps0/18685172aa2c45a7295d877b3c9a1206).

**Comment by Coder-256** [comment]
@1ps0 I might want to look into this more. It probably would fix #398 and maybe even #332!

**Comment by Coder-256** [comment]
I haven't actually done this because the Objective-C backend (which must somehow communicate with the main application while also being called at any time and taking STDIN) is beyond my abilities. However, I would be able to do the Javascript if somebody was willing to do the Objective-C.

However, a Safari extension with much shared code could also be made, and the Objective-C bridge would be [much easier](https://developer.apple.com/library/safari/documentation/Tools/Conceptual/SafariExtensionGuide/CommunicatingwithyourOSXApplication/CommunicatingwithyourOSXApplication.html). In addition, this would fix #402 without needing to use the `Developer` menu, which is not ideal for end users.

Over all, browser extensions seem to be a better way to go. Also, they would still work if BeardedSpice ever became cross-platform (I have no idea if that is in mind, though; @Stillness-2: is it?)

**TL;DR** I would be willing to collaborate to work on browser extensions because they have more pros over  using Apple events and the broken Scripting Bridge. However, doing this is a large undertaking.

**Comment by robrecord** [comment]
May I +1 this?
